File: C/kazuate.c
#include<stdio.h>
#include<stdlib.h>
#include<time.h>
int main(){ 
  int ans_num,usr_ans,cnt=0;
  srand((unsigned)time(NULL)); 
  ans_num=rand()%100+1;
  puts("1から100までの数あてゲームです");

  do{ 
      cnt++;
      puts("数値を入力してください");
      scanf("%d",&usr_ans);
      if(ans_num==usr_ans){
          puts("正解です");
      }else if(ans_num<usr_ans){
          puts("もっと小さいです");
      } 
      else if(ans_num>usr_ans){
          puts("もっと大きいです");
      }
  }while(ans_num!=usr_ans);
printf("%d回であたりました",cnt);
return 0;
}

File: ruby/kazuate.rb
ans_num=rand(100)+1
usr_ans=nil
count=0
puts'1から100で数あてゲームです'
while usr_ans!=ans_num do 
    usr_ans=gets.to_i
    if usr_ans==ans_num
        puts " 正解です"
    elsif  usr_ans>ans_num
        puts "もっと小さいよ！"
    else usr_ans<ans_num 
        puts "もっと大きいよ！" 
    end
    count+=1
end
puts count.to_s+"回であたりました！！"
